test/system/functional_correctness_test.py

```diff
@@ -0,0 +1,162 @@
+    os.makedirs(os.path.dirname(temp_input_path), exist_ok=True)
+    with open(temp_input_path, "w") as f:
+        f.writelines([f"chronicle_0_0.story_0_0.{i}\n" for i in range(1, 101)])
+    print("Input file generated")
+
+
+# Deployment and System Control Functions
+def deploy_system():
+    """
+    Deploys the system.
+    """
+    deploy_command = [
+        os.path.join(DEPLOY_PATH, "build_install_deploy.sh"), "-type", "Release", "-n", "5", "-j", "2", "-d"
+    ]
+    run_command(deploy_command)
+
+
+def run_client():
+    """
+    Runs the client admin process in a Spack environment.
+    """
+    generate_input_file()
+    client_admin_command = f"""
+        spack env activate -p {REPO_PATH} &&
+        export LD_LIBRARY_PATH={os.path.join(INSTALL_PATH, "lib")}:$LD_LIBRARY_PATH &&
+        {os.path.join(BIN_PATH, "client_admin")} --config {os.path.join(CONF_PATH, "client_conf.json")} -f {os.path.join(BIN_PATH, "temp_input")}
+    """
+    run_command(client_admin_command, shell=True)
+
+
+def stop_execution():
+    """
+    Stops the execution of the system.
+    """
+    single_user_command = [os.path.join(DEPLOY_PATH, "local_single_user_deploy.sh"), "-s", "-w", INSTALL_PATH]
+    run_command(single_user_command)
+
+
+# Test Functions
+def report_result(test_name, description, result, is_success):
+    highlight = GREEN_HIGHLIGHT if is_success else RED_HIGHLIGHT
+    status = "Success" if is_success else "Failure"
+    print(f"{test_name}:\n***{description}***\n{highlight}{status}{RESET}\n")
+
+
+def test_message_storage_validation():
+    """
+    Checks that each file entry contains the specified pattern.
+    """
+    file_list = glob.glob(os.path.join(OUTPUT_PATH, "chronicle_0_0.story_0_0.*.csv"))
+    if not file_list:
+        return "Test Failure: No matching files found.", False
+
+    for file in file_list:
+        with open(file, "r") as f:
+            if any("chronicle_0_0.story_0_0" not in line for line in f):
+                return f"Test Failure: Entry without 'chronicle_0_0.story_0_0' found in {file}.", False
+
+    return "Test Success: All entries contain 'chronicle_0_0.story_0_0'.", True
+
+def test_timestamp_range_check():
+    """
+    Ensures all timestamps fall within the expected start and end range for each file.
+    """
+    file_list = glob.glob(os.path.join(OUTPUT_PATH, "chronicle_0_0.story_0_0.*.csv"))
+    if not file_list:
+        return "Test Failure: No matching files found.", False
+
+    for file in file_list:
+        filename_parts = os.path.basename(file).split('.')
+        try:
+            story_chunk_start = int(filename_parts[2]) * 1_000_000_000
+        except (IndexError, ValueError):
+            print(f"Skipping {file} due to start time extraction issue.")
+            continue
+
+        with open(file, "r") as f:
+            lines = f.readlines()
+            first_timestamp = int(lines[0].strip().split(":")[2]) if lines else None
```

Comment:
This assumes a correct ordering of events. If a csv file has correct range but is not ordered correctly, the first and last event cannot be used as the timestamp boundaries.

Reply:
@fkengun  Please, note that the way the test is written first_timestamp is the timestamp of the event on the first line, not the story_chunk_start, and the last_timestamp is the timestamp of the event on the last line not story_chunk_end 

the conditions that have to hold for any story_chunk file:
1. any_timestamp >= previous_line_timestamp
2. first_timestamp >= story_chunk_start
3. last_timestamp < next_story_chunk_start

Reply:
@ibrodkin Condition 1 is about sorting. Only condition 2 and 3 are about timestamp range. I understand the first and last timestamps are not story chunk boundaries. Since I do not see a good way of capturing the start and end time of a story chunk into the csv files because they are calculated in Keepers, I do not see a good way of only verifying event timestamp ranges in csv files. I think we cannot separate timestamp range check and ordering check for csv files. We can do it in HDF5 files when we also write down story chunk infos. @EnekoGonzalez3

Reply:
Understood. But there is no story_chunk_end. So we want to use the next story's story_chunk_start? How can this be forced?

Reply:
@EnekoGonzalez3 I think you wanted to say next story CHUNK's `story_chunk_start`. `StoryChunk` has `start_time` and `end_time` https://github.com/grc-iit/ChronoLog/blob/6b7a6849801caac1208b0d23327d84cf240bb565/chrono_common/StoryChunk.h#L111. We can make it work by including `start_time` and `end_time` to the name of the file in https://github.com/grc-iit/ChronoLog/blob/6b7a6849801caac1208b0d23327d84cf240bb565/ChronoGrapher/CSVFileChunkExtractor.cpp#L26. My point was there is no way to do boundary checks with existing codes. But if we update it a bit, it is doable.

Reply:
Oh yes. That's what I meant, and by saying there is no end time, i meant from my test point of view. If we add the end time, then there is no problem.

Reply:
see my next comment to answer the questions, i actually meant to place the comment there :)

the story chunk boundaries are encoded in the file names (note that the boundary is guaranteed to be on the second boundary by design). current file name gives you the story_chunk_start boundary in seconds, and the next file name gives you the stroy_chunk_end boundary in seconds. You are controlling the client producing the events, so you can guarantee that there are no gaps between the events greater that the second, so there should not be any empty story_chunks in the chrono_grapher memory that would not produce the csv file. Thus the filename of the next file can used as a source of the story_chunk_end.  
Also note about restructuring the test so that you have only file reading loop in the main routine of your test. you can retain the last_event_timestamp from the previous file tested and do the check for previous_file_last_event_timestamp < current_chunk_start timestamp when the next file is loaded.  you can also retain the name of the previous file so that in the check fails  tou can report the file that failed the test correctly.